flight/Libraries/math/pid.c

```diff
@@ -199,6 +230,67 @@ void pid_configure(struct pid *pid, float p, float i, float d, float iLim)
 }
 
 /**
+ * Assigns a pid_deadband struct to the PID controller.
+ * @param[out] pid The PID structure to configure
+ * @param[in] deadband The external deadband structure to assign
+ */
+void pid_assign_deadband(struct pid *pid, struct pid_deadband *db)
+{
```

Comment:
Hmm, the pointer assignment stuff is there to be able to disable the deadband between mode switches, while still being able to update it with no fuzz via the settings callback (maybe allow it as option in TxPID). I suppose I could put a check around the pid_configure_deadband calls and just not malloc memory, if all deadbands are zero. If there's no deadband, it'll assign NULL to begin with when passing the pointer along.

That said, the assign and remove functions are mostly for readability and unneeded abstraction, we might aswell turn them into macros in stab.c.

If we malloc for saving RAM, maybe we should also cause a system configuration error on F1 when any deadbands are enabled the same time as autotune.

On further note, thanks to some shower thoughts, I guess it makes more sense to pass the deadband via argument to pid_apply_setpoint (which the rate controllers use), that's four bytes on the call stack. Like this we can scrub the pointer from the struct. Also, would we ever need deadbands in the other two, anyway? If not, I'll remove the calls and make some codespace for the checks above.

I'll write a commit with those things for review this evening when I'm back for work.